opencti-platform/opencti-front/src/private/components/settings/exclusion_lists/ExclusionListValidator.ts

```diff
@@ -1,10 +1,20 @@
 import * as Yup from 'yup';
 
-const exclusionListValidator = (t: (value: string) => string) => Yup.object().shape({
-  name: Yup.string().trim().min(2).required(t('This field is required')),
-  description: Yup.string().nullable(),
-  exclusion_list_entity_types: Yup.array().min(1, t('Minimum one entity type')).required(t('This field is required')),
-  file: Yup.mixed().required(t('This field is required')),
-});
+const exclusionListValidator = (t: (value: string) => string, isCreatedWithFile: boolean) => {
+  if (isCreatedWithFile) {
+    return Yup.object().shape({
+      name: Yup.string().trim().min(2).required(t('This field is required')),
+      description: Yup.string().nullable(),
+      exclusion_list_entity_types: Yup.array().min(1, t('Minimum one entity type')).required(t('This field is required')),
+      file: Yup.mixed().required(t('This field is required')),
```

Comment:
Instead of completely rewriting the object you could do something like :
```
file: isCreatedWithFile ? Yup.mixed().required(t('This field is required')) : Yup.mixed(),
content: isCreatedWithFile ? Yup.string() : Yup.string().required(t('This field is required')),
```

Reply:
Done